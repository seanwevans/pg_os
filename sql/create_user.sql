\set ECHO none
\i :abs_srcdir/../usersrolespermissions.sql
\set ECHO queries
\set VERBOSITY terse

-- successful creation
SELECT create_user('alice') AS user_id;

\set ON_ERROR_STOP off
-- duplicate username should fail
SELECT create_user('alice');

-- null username should fail
SELECT create_user(NULL);

-- empty username should fail
SELECT create_user('');
\set ON_ERROR_STOP on
ALTER SEQUENCE users_id_seq RESTART WITH 2;
