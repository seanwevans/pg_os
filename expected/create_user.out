\set ECHO none
SELECT create_user('alice') AS user_id;
 user_id 
---------
       1
(1 row)

SELECT create_user('alice');
ERROR:  duplicate key value violates unique constraint "users_username_key"
SELECT create_user(NULL);
ERROR:  username cannot be empty
SELECT create_user('');
ERROR:  username cannot be empty
ALTER SEQUENCE users_id_seq RESTART WITH 2;
